File: cypress-framework/cypress/pages/ProductPage.js
class ProductPage {
  static visit(id) { cy.visit(`/index.php?route=product/product&product_id=${id}`); return this; }
  static setQuantity(q) { cy.get("#input-quantity").clear().type(String(q)); return this; }
  static addToCart() { cy.get("#button-cart").click(); return this; }
  static successToast() { return cy.get(".alert-success, #alert .alert-success"); }
}
export default ProductPage;

File: selenium-framework/pages/product_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class ProductPage(BasePage):
    ADD_TO_CART = (By.ID, "button-cart")
    QUANTITY    = (By.ID, "input-quantity")
    SUCCESS     = (By.CSS_SELECTOR, ".alert-success, #alert .alert-success")
    NAME        = (By.CSS_SELECTOR, "h1")

    def go(self, product_id: int):
        return self.open(f"product/product&product_id={product_id}")

    def add_to_cart(self, qty: int = 1):
        self.type(self.QUANTITY, str(qty))
        self.click(self.ADD_TO_CART)
        return self

    def success_visible(self) -> bool:
        return self.is_visible(self.SUCCESS)
